jwst/ramp_fitting/ramp_fit.py

```diff
@@ -632,85 +659,152 @@ def ols_ramp_fit(data, err, groupdq, inpixeldq, buffsize, save_opt, readnoise_2d
         actual_segments = 0
         actual_CRs = 0
 
-    return new_model.data, new_model.dq, new_model.var_poisson, new_model.var_rnoise, new_model.err, \
+    return new_model.data, new_model.dq, new_model.var_poisson, \
+        new_model.var_rnoise, new_model.err, \
         int_data, int_dq, int_var_poisson, int_var_rnoise, int_err, int_int_times, \
         opt_slope, opt_sigslope, opt_var_poisson, opt_var_rnoise, opt_yint, opt_sigyint, \
         opt_pedestal, opt_weights, opt_crmag, actual_segments, actual_CRs
 
 
-def ramp_fit_miri_multigroups(data, err, groupdq, cubeshape, imshape, nreads, ngroups):
+def ols_ramp_fit_single(
+        input_model, int_times, buffsize, save_opt, readnoise_2d, gain_2d, weighting):
     """
-    For MIRI datasets having >1 group, if all pixels in the final group are
-    flagged as DO_NOT_USE, resize the input model arrays to exclude the
-    final group.  Similarly, if leading groups 1 though N have all pixels
-    flagged as DO_NOT_USE, those groups will be ignored by ramp fitting, and
-    the input model arrays will be resized appropriately. If all pixels in
-    all groups are flagged, return None for the models.
+    Fit a ramp using ordinary least squares. Calculate the count rate for each
+    pixel in all data cube sections and all integrations, equal to the weighted
+    slope for all sections (intervals between cosmic rays) of the pixel's ramp
+    divided by the effective integration time.
 
-    Parameters
-    ----------
-    data: ndarray
-        4-D image cube with dimensions (integrations, groups, rows, columns)
+    Parameter
+    ---------
+    input_model: RampModel
 
-    err: ndarray
-        Error array
+    int_times : None
+        Not used
```

Comment:
I didn't remove it because I wasn't sure why it was here to begin with.  I will remove it if I determine it's not actually used or can easily be gotten from the classes.